### ISSUE TITLE
Overflowing Int literals

### ISSUE THREAD

**Issue by stefan-hoeck** [idris-lang/Idris2#1101]
Seems like `Int` literals are not properly truncated (the BitsXY primitives seem to work correctly).

# Steps to Reproduce

Run

```idris
main : IO ()
main = do printLn (the Int 10000000000000000000000000000)
          printLn (the Int 1 * 10000000000000000000000000000)
```

# Expected Behavior

Prints
```
4477988020393345024
4477988020393345024
```

# Observed Behavior

Prints
```
10000000000000000000000000000
4477988020393345024
```

**Comment by gallais** [comment]
This is related to #1048: `Int` have no set meaning AFAIK and the
implementation is backend-dependent. This is obviously a bad situation
to be in and we should introduce `Int32` and `Int64` and decide which
one `Int` is an alias for.

**Comment by stefan-hoeck** [comment]
OK, thanks. Look, I could give this a try and add Int32 and Int64 given that I have already been working on primitives regarding their bitops. However, I'm not sure I'm up to the task and will definitely need some guidance along the way (especially with the C backend: I am not much of a C programmer). Or is this already planned and scheduled by the core Idris team (in which case I'll happily step aside, :-) )?

Personally, I'd opt for making Int an alias for Int64 as this is how it is handled right now in the JS and Scheme backends. In JS, Int32 could be represented by `Number` (thus resolving #801) and Int64 by BitInt (as Int is right now).

**Comment by stefan-hoeck** [comment]
Fixed in #1211
